src/nifreeze/model/base.py

```diff
@@ -78,114 +79,81 @@ class BaseModel:
 
     """
 
-    __slots__ = (
-        "_model",
-        "_mask",
-        "_models",
-        "_datashape",
-        "_is_fitted",
-        "_modelargs",
-    )
+    __slots__ = {
+        "_dataset": "Reference to a :obj:`~nifreeze.data.base.BaseDataset` object.",
+    }
```

Comment:
For ``__slots__`` to be documented with autodoc you write them as a dict.

Reply:
Why document a private variable?

Reply:
I just left private variable comments on the DWI base class because that'd make it easier to implement models in the future.